lib/client/index.js

```diff
@@ -0,0 +1,123 @@
+ *
+ * @param _$fh - The feedhenry client API.
+ * @param _syncOptions - The options to configure the sync client. See link above for available options.
+ * @param _mediator - The mediator to use for publishing topics.
+ */
+function init(_$fh, _syncOptions, _mediator) {
+  if (initialized) {
+    console.log('sync-client already initalized.');
+    return;
+  }
+
+  //Assigning a single mediator for the module.
+  //All data managers will use the same mediator to publish topics to.
+  mediator = _mediator;
+
+  $fh = _$fh;
+
+  //Adding a listener that will be executed every time the notify handler is executed.
+  //This listener function will send the syncNotification object (see documentation above) to the syncNotification stream, which in turn will propigate though any
+  //other streams attached to it.
+  function addObserverStreamListener(syncNotificationsObserver) {
+    addListener(function(syncNotification) {
+      syncNotificationsObserver.onNext(syncNotification);
+    });
+  }
+
+  //Creating a single notification stream that will be used to handle all notifications from the $fh.sync API.
+  syncNotificationStream = Rx.Observable.create(addObserverStreamListener).share();
+
+  var syncOptions = _.defaults(_syncOptions, defaultConfig.syncOptions);
+
+  //Initialising the $fh.sync API with the required options.
+  $fh.sync.init(syncOptions);
+  initialized = true;
+
+  //Registering a notification listener with the $fh.sync API. This will be called for all data sets managed by the
+  //sync framework.
+  $fh.sync.notify(function(syncNotification) {
+    syncNotifyListeners.forEach(function(syncNotifyListener) {
+      syncNotifyListener.call(undefined, syncNotification);
+    });
+  });
+}
+
+
+/**
+ *
+ * Function to create a manager for a single data set (e.g. workorders, messages etc).
+ *
+ * This will use the $fh.sync.manage API call to manage a specific data set. In addition, another Observable stream
+ * will be created to filter out notification messages for messages that are only relevant to this data set.
+ *
+ * @param datasetId - An identifier for the data set (E.g. workorders, results etc)
+ * @param options - Data Set specific options for the data set (See $fh.sync documentation linked above for more information on data set parameters)
+ * @param queryParams - Parameters to pass when performing sync operations (See $fh.sync documentation linked above for more information on data set query parameters)
+ * @param metaData - Any metadata to be passed with the sync operations. (See $fh.sync documentation linked above for more information on data set query parameters)
+ * @returns {*|promise}
+ */
+function manage(datasetId, options, queryParams, metaData) {
+  var deferred = q.defer();
+
+  //The $fh.sync API needs to be initialised before a data set can be managed.
+  if (!initialized) {
+    deferred.resolve('Sync not yet initialized.  Call sync-client.init() first.');
```

Comment:
Wouldn't this generally be a bug?

At the very least, this requires anyone using this api to then do an `instanceof String` check on the result, and if it's a string, to handle it as an error.